### PR TITLE
add more tests for HTTPS

### PR DESCRIPTION
**Pull request by Ladicek**
Specifically, this commit adds tests for:

- 1way SSL, using legacy security
- 2way SSL, using legacy security
- 2way SSL with authorization, using legacy security
- 1way SSL, using Elytron
- 2way SSL, using Elytron
- 2way SSL with authorization, using Elytron

### CODE REVIEW

File: protocols/https-1way-elytron/pom.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                        <id>server-certificate</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>exportCertificate</goal>
+                        </goals>
+                        <configuration>
+                            <keystore>${project.build.directory}/server-keystore.jks</keystore>
+                            <storepass>server-password</storepass>
+                            <alias>server-key</alias>
+                            <keypass>server-password</keypass>
```
Comment (penehyba) [inline protocols/https-1way-elytron/pom.xml]: keypass in <goal>exportCertificate</goal> is unused. There are 12 occurrences (IDEA identifies it as "Element keypass is not allowed here").

File: protocols/https-2way-authz-elytron/src/test/java/org/wildfly/swarm/ts/protocols/https/twoway/authz/elytron/Https2wayAuthzElytronTest.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    @RunAsClient
+    public void http() throws IOException {
+        String response = Request.Get("http://localhost:8080/").execute().returnContent().asString();
+        assertThat(response).isEqualTo("Hello on port 8080, secure: false, principal: null, isGuest: false, isUser: false");
+    }
+
+    @Test
+    @RunAsClient
+    public void https_authenticatedAndAuthorizedClient() throws IOException, GeneralSecurityException {
+        char[] clientPassword = "client-password".toCharArray();
```
Comment (penehyba) [inline protocols/https-2way-authz-elytron/src/test/java/org/wildfly/swarm/ts/protocols/https/twoway/authz/elytron/Https2wayAuthzElytronTest.java]: just suggestion: private static final CLIENT_PWD,
 createCustomSSLContext("keystore.jks", "truststore.jks", pwd)
 createCustomClient(context, nameVerifier, keyMaterial, trustMaterial)
These parts repeat in every test, anyway I didn't think of any better solution (builder pattern, abstract ancestor..)

File: protocols/https-1way-elytron/pom.xml (hunk 1 above)
Comment (Ladicek) [reply protocols/https-1way-elytron/pom.xml]: It's unfortunate that `keytool` accepts `-keypass` here, but the Maven plugin doesn't expose that option. I'll remove it.

File: protocols/https-2way-authz-elytron/src/test/java/org/wildfly/swarm/ts/protocols/https/twoway/authz/elytron/Https2wayAuthzElytronTest.java (hunk 2 above)
Comment (Ladicek) [reply protocols/https-2way-authz-elytron/src/test/java/org/wildfly/swarm/ts/protocols/https/twoway/authz/elytron/Https2wayAuthzElytronTest.java]: I know there's a bit of repetition here, but I think the Apache HttpClient API is so succinct it doesn't make too much sense to abstract over it again. I think I prefer to have all the code on a single place here.

**Pull request merged**